addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/MiHomeSubdevicesHandler.java

```diff
@@ -0,0 +1,651 @@
+
+            this.mihomeID = deviceID;
+
+            JsonObject data = gatewayHandler.getSubdeviceData(this.mihomeID);
+            if (data != null) {
+                updateThingProperties(data);
+
+                updateStatus(ThingStatus.ONLINE);
+
+                scheduleRegularUpdate(updateInterval);
+            } else {
+                logger.warn("Mi|Home server may be temporary unavailable or device with ID {} may be deleted.",
+                        this.mihomeID);
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "Missing data for paired device with ID " + this.mihomeID);
+            }
+        } else {
+            schedulePairing(gatewayHandler.getGatewayId());
+        }
+    }
+
+    private boolean isGatewayReady() {
+        return gatewayHandler.getGatewayId() > 0;
+    }
+
+    private Integer getDeviceID() {
+        Integer id = null;
+        // The thing has been once initialized and the deviceID has been persisted
+        logger.debug("Searching deviceID in the thing configuration.");
+        Configuration config = getConfig();
+        BigDecimal deviceID = (BigDecimal) config.get(MiHomeBindingConstants.PROPERTY_DEVICE_ID);
+        if (deviceID != null) {
+            id = deviceID.intValue();
+            return id;
+        }
+
+        // The thing has been created by a DiscoveryService and the deviceID is included in the properties
+        logger.debug("Searching device ID in the thing properties.");
+        Map<String, String> properties = editProperties();
+        if (properties.containsKey(MiHomeBindingConstants.PROPERTY_DEVICE_ID)) {
+            String propertyValue = properties.get(MiHomeBindingConstants.PROPERTY_DEVICE_ID);
+            try {
+                id = Integer.parseInt(propertyValue);
+            } catch (NumberFormatException e) {
+                logger.debug("Cann't parse property {} as int, value is {}", MiHomeBindingConstants.PROPERTY_DEVICE_ID,
+                        propertyValue);
+            }
+        }
+        return id;
+    }
+
+    private void schedulePairing(final int gatewayID) {
+        if (pairingTask != null && !pairingTask.isDone()) {
+            logger.info("Pairing in progress !");
+            return;
+        }
+        ThingTypeUID thingTypeUID = getThing().getThingTypeUID();
+        final String type = MiHomeBindingConstants.THING_TYPE_TO_DEVICE_TYPE.get(thingTypeUID);
+
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    for (int i = 0; i < PAIRING_RETRY_COUNT; i++) {
+                        logger.debug("Trying to pair the device of type {}, attempt {}", type, i + 1);
+                        if (pairDevice(gatewayID, type)) {
+                            break;
+                        }
+                    }
+                } catch (Exception e) {
```

Comment:
We should still be more explicit about what might have gone wrong, rather than catching everything. We want to avoid catching errors that are completely unrecoverable anyway.